credentials/credentials.go

```diff
@@ -158,11 +158,20 @@ func ParseSingleCredentials(credentialsMap map[string]interface{}) (Credentials,
 	default:
 		return nil, fmt.Errorf("entry %s: unknown credentials type: %s", id, credentialsType)
 	}
-	err := mapstructure.Decode(credentialsMap, credentials)
+	var validationErrors error
+	delete(credentialsMap, "type")
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		ErrorUnused: true,
+		Metadata:    nil,
+		Result:      credentials})
 	if err != nil {
+		validationErrors = multierror.Append(validationErrors, fmt.Errorf("entry %s: unable to create a decoder: %v", id, err))
```

Comment:
As of go 1.20, they added this to the std lib. You can simply use [this](https://pkg.go.dev/errors#example-Join). If this project is not at version 1.20, then this is fine unless you want to update the version of go.

Reply:
Okay, thought the interface was kinda the same. In any case `multierror` is fine.